src/ProductConstructionService/ProductConstructionService.Api/Queue/Jobs/Job.cs

```diff
@@ -0,0 +1,12 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Text.Json.Serialization;
+
+namespace ProductConstructionService.Api.Queue.Jobs;
+
+[JsonDerivedType(typeof(TextJob), typeDiscriminator: nameof(TextJob))]
+public abstract class Job
+{
+    public Guid Id { get; set; } = Guid.NewGuid();
```

Comment:
Should this also be `required`?

Reply:
I added `required` and `init` to it. I think it's better that way, as the Id won't be able to change, but we'll have to set the Id on the producer side